### ISSUE TITLE
Error running update command when migrating to v6

### ISSUE THREAD

**Issue by andrewseguin** [angular/angular-cli#10363]
### Versions
```
  

Angular CLI: 6.0.0-rc.4
Node: 8.9.4
OS: darwin x64
Angular: 6.0.0-rc.5
... animations, common, compiler, compiler-cli, core, forms
... http, platform-browser, platform-browser-dynamic, router

Package                      Version
------------------------------------------------------
@angular-devkit/architect    0.5.6
@angular-devkit/core         0.5.6
@angular-devkit/schematics   0.5.6
@angular/cdk                 5.0.2
@angular/cli                 6.0.0-rc.4
@angular/material            5.0.2
@angular/tsc-wrapped         4.4.6
@ngtools/json-schema         1.1.0
@schematics/angular          0.5.6
@schematics/update           0.5.6
rxjs                         5.6.0-forward-compat.4
typescript                   2.4.2

```

### Repro steps

* Step 1 - Update project's dependencies from Angular v5 and CLI 1.6.1 to Angular v6 and CLI v6
* Step 2 - Run `npm i`
* Step 3 - Observe message:
```
> @angular/cli@6.0.0-rc.4 postinstall /Users/andrewjs/git/beautiful-day/node_modules/@angular/cli
> node ./bin/ng-update-message.js


================================================================================
An old CLI configuration has been detected, which needs to be updated to the latest version.

Please run the following command to update this workspace:
  ng update @angular/cli --migrate-only --from=1
================================================================================
```
* Step 4 - Run provided command: `ng update @angular/cli --migrate-only --from=1`

### Observed behavior
Received error:
```
^_^   ng update @angular/cli --migrate-only --from=1

Schematic input does not validate against the Schema: {"dryRun":false,"migrateOnly":true,"from":1,"force":false,"all":false,"next":false,"registry":"http://registry.npmjs.org/"}
Errors:

  Data path ".from" should be string.
```

### Desired behavior
Should migrate my project

**Comment by andrewseguin** [comment]
Unblocked by changing command to be `from:1.6.1` instead of `from:1`.

**Comment by andrewseguin** [comment]
Just kidding, here's the output of running `ng update @angular/cli --migrate-only --from=1.6.1`

```
^_^   ng update @angular/cli --migrate-only --from=1.6.1
            Updating karma configuration
            Updating configuration
            Removing old config file (angular-cli.json)
            Writing config file (angular.json)

> node-sass@4.8.3 install /Users/andrewjs/git/beautiful-day/node_modules/node-sass
> node scripts/install.js

Cached binary found at /Users/andrewjs/.npm/node-sass/4.8.3/darwin-x64-57_binding.node

> node-sass@4.8.3 postinstall /Users/andrewjs/git/beautiful-day/node_modules/node-sass
> node scripts/build.js

Binary found at /Users/andrewjs/git/beautiful-day/node_modules/node-sass/vendor/darwin-x64-57/binding.node
Testing binary
Binary is fine
npm WARN @angular/cdk@5.0.2 requires a peer of @angular/core@~5.1.0 but none is installed. You must install peer dependencies yourself.
npm WARN @angular/cdk@5.0.2 requires a peer of @angular/common@~5.1.0 but none is installed. You must install peer dependencies yourself.
npm WARN @angular/common@6.0.0-rc.5 requires a peer of rxjs@^6.0.0-rc.0 but none is installed. You must install peer dependencies yourself.
npm WARN @angular/compiler-cli@6.0.0-rc.5 requires a peer of typescript@>=2.7.2 <2.8 but none is installed. You must install peer dependencies yourself.
npm WARN @angular/core@6.0.0-rc.5 requires a peer of rxjs@^6.0.0-rc.0 but none is installed. You must install peer dependencies yourself.
npm WARN @angular/core@6.0.0-rc.5 requires a peer of zone.js@~0.8.26 but none is installed. You must install peer dependencies yourself.
npm WARN @angular/forms@6.0.0-rc.5 requires a peer of rxjs@^6.0.0-rc.0 but none is installed. You must install peer dependencies yourself.
npm WARN @angular/http@6.0.0-rc.5 requires a peer of rxjs@^6.0.0-rc.0 but none is installed. You must install peer dependencies yourself.
npm WARN @angular/material@5.0.2 requires a peer of @angular/core@~5.1.0 but none is installed. You must install peer dependencies yourself.
npm WARN @angular/material@5.0.2 requires a peer of @angular/common@~5.1.0 but none is installed. You must install peer dependencies yourself.
npm WARN @angular/router@6.0.0-rc.5 requires a peer of rxjs@^6.0.0-rc.0 but none is installed. You must install peer dependencies yourself.
npm WARN angularfire2@5.0.0-rc.3 requires a peer of @angular/common@^4.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN angularfire2@5.0.0-rc.3 requires a peer of @angular/core@^4.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN angularfire2@5.0.0-rc.3 requires a peer of @angular/platform-browser@^4.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN angularfire2@5.0.0-rc.3 requires a peer of @angular/platform-browser-dynamic@^4.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN angularfire2@5.0.0-rc.3 requires a peer of rxjs@^5.0.1 but none is installed. You must install peer dependencies yourself.
npm WARN codelyzer@4.0.1 requires a peer of @angular/compiler@^2.3.1 || >=4.0.0-beta <6.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN codelyzer@4.0.1 requires a peer of @angular/core@^2.3.1 || >=4.0.0-beta <6.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN codelyzer@4.0.1 requires a peer of @angular/platform-browser-dynamic@^2.3.1 || >=4.0.0-beta <6.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN codelyzer@4.0.1 requires a peer of @angular/platform-browser@^2.3.1 || >=4.0.0-beta <6.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN codelyzer@4.0.1 requires a peer of @angular/common@^2.3.1 || >=4.0.0-beta <6.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN @ngtools/webpack@6.0.0-rc.4 requires a peer of typescript@~2.5.0 || ~2.6.0 || ~2.7.0 but none is installed. You must install peer dependencies yourself.
npm WARN ajv-keywords@3.1.0 requires a peer of ajv@^6.0.0 but none is installed. You must install peer dependencies yourself.

added 700 packages in 22.502s
DELETE angular-cli.json
CREATE angular.json (6416 bytes)
UPDATE karma.conf.js (1021 bytes)
UPDATE package.json (2014 bytes)
/Users/andrewjs/git/beautiful-day/node_modules/@angular-devkit/schematics/node_modules/rxjs/internal/util/subscribeTo.js:42
        throw new TypeError(msg);
        ^

TypeError: You provided 'function reduceOperatorFunction(source) {
        return pipe_1.pipe(scan_1.scan(function (acc, value, index) {
            return accumulator(acc, value, index + 1);
        }), takeLast_1.takeLast(1))(source);
    }' where a stream was expected. You can provide an Observable, Promise, Array, or Iterable.
    at Object.exports.subscribeTo (/Users/andrewjs/git/beautiful-day/node_modules/@angular-devkit/schematics/node_modules/rxjs/internal/util/subscribeTo.js:42:15)
    at Object.subscribeToResult (/Users/andrewjs/git/beautiful-day/node_modules/@angular-devkit/schematics/node_modules/rxjs/internal/util/subscribeToResult.js:7:26)
    at MergeMapSubscriber._innerSub (/Users/andrewjs/git/beautiful-day/node_modules/@angular-devkit/schematics/node_modules/rxjs/internal/operators/mergeMap.js:132:38)
    at MergeMapSubscriber._tryNext (/Users/andrewjs/git/beautiful-day/node_modules/@angular-devkit/schematics/node_modules/rxjs/internal/operators/mergeMap.js:129:14)
    at MergeMapSubscriber._next (/Users/andrewjs/git/beautiful-day/node_modules/@angular-devkit/schematics/node_modules/rxjs/internal/operators/mergeMap.js:112:18)
    at MergeMapSubscriber.notifyComplete (/Users/andrewjs/git/beautiful-day/node_modules/@angular-devkit/schematics/node_modules/rxjs/internal/operators/mergeMap.js:148:18)
    at InnerSubscriber._complete (/Users/andrewjs/git/beautiful-day/node_modules/@angular-devkit/schematics/node_modules/rxjs/internal/InnerSubscriber.js:37:21)
    at InnerSubscriber.Subscriber.complete (/Users/andrewjs/git/beautiful-day/node_modules/@angular-devkit/schematics/node_modules/rxjs/internal/Subscriber.js:128:18)
    at Observable.rxjs_1.concat.rxjs_1.concat.rxjs_1.Observable.obs [as _subscribe] (/Users/andrewjs/git/beautiful-day/node_modules/@angular-devkit/schematics/tools/workflow/node-workflow.js:113:17)
    at Observable.subscribe (/Users/andrewjs/git/beautiful-day/node_modules/@angular-devkit/schematics/node_modules/rxjs/internal/Observable.js:162:69)
```

**Comment by cyrilletuzi** [comment]
Both issues (`--from` error and `Observable` errors) are fixed in CLI rc5, just update your CLI.

**Comment by filipesilva** [comment]
The first problem (`Schematic input does not validate`) is reported as https://github.com/angular/angular-cli/issues/10312 and should be fixed in rc.5. 

The second problem (`TypeError: You provided 'function reduceOperatorFunction`) is reported as https://github.com/angular/angular-cli/issues/10335 and fixed as of rc.5.

If either is not working, we should reopen those tracking issues.

The parenthesis problem seems fixed on master.

**Comment by danboh** [comment]
Hi! I'm new to Angular and was working on a starter project that used 5.2, but it seems that after running 
`npm install -g npm-check-updates`
`ng update @angular/cli`

Everything was updated to the latest version (6.0) and now getting the following error when running ng serve

> ERROR in The Angular Compiler requires TypeScript >=2.7.2 and <2.8.0 but 2.8.3 was found instead.


I tried updating `package.json` to  `"typescript": "2.8.0"` but it doesn't seem to do the trick. 

Is this an issue with CLI? Apologies if this is a really lame question but just learning 😃
